pallets/manta-pay/src/lib.rs

```diff
@@ -1402,3 +1406,278 @@ where
         Ok(())
     }
 }
+
+/// This allows `ToPrivate` to insert arbitrary UTXOs. There are no checks on the source accounts. `ToPublic` and `PrivateTransfer` will fail.
+///
+/// WARNING: Ensure that only `AssetId` that corresponds to an SBT are allowed to call this Ledger. It will allow arbitrary UTXO to be inserted.
+struct SBTLedger<T>(PhantomData<T>)
+where
+    T: Config;
+
+impl<T> SenderLedger<config::Parameters> for SBTLedger<T>
+where
+    T: Config,
+{
+    type SuperPostingKey = (Wrap<()>, ());
+    type ValidUtxoAccumulatorOutput = Wrap<config::UtxoAccumulatorOutput>;
+    type ValidNullifier = Wrap<config::Nullifier>;
+    type Error = SenderLedgerError;
+
+    #[inline]
+    fn is_unspent(
+        &self,
+        nullifier: config::Nullifier,
+    ) -> Result<Self::ValidNullifier, Self::Error> {
+        if NullifierCommitmentSet::<T>::contains_key(
+            fp_encode(nullifier.nullifier.commitment).map_err(SenderLedgerError::FpEncodeError)?,
+        ) {
+            Err(SenderLedgerError::AssetSpent)
+        } else {
+            Ok(Wrap(nullifier))
+        }
+    }
+
+    #[inline]
+    fn has_matching_utxo_accumulator_output(
+        &self,
+        output: config::UtxoAccumulatorOutput,
+    ) -> Result<Self::ValidUtxoAccumulatorOutput, Self::Error> {
+        let accumulator_output = fp_encode(output).map_err(SenderLedgerError::FpEncodeError)?;
+        // NOTE: Checking for an empty(zeroed) byte array. This happens for UTXOs with `value = 0`,
+        // for which you dont need a membership proof, but you still need a root (in this case
+        // zeroed).
+        if accumulator_output == [0u8; 32]
+            || UtxoAccumulatorOutputs::<T>::contains_key(accumulator_output)
+        {
+            return Ok(Wrap(output));
+        }
+        Err(SenderLedgerError::InvalidUtxoAccumulatorOutput)
+    }
+
+    #[inline]
+    fn spend_all<I>(
+        &mut self,
+        _super_key: &Self::SuperPostingKey,
+        _iter: I,
+    ) -> Result<(), Self::Error>
+    where
+        I: IntoIterator<Item = (Self::ValidUtxoAccumulatorOutput, Self::ValidNullifier)>,
+    {
+        Ok(())
```

Comment:
why we put a dummy implementation here? Just feel a little confused by this function name.

Reply:
It is a trait that needs to be implemented from `manta-rs`. Hence the dummy implementation